### PR TITLE
Add GCS input to Cloudflare Logpush

### PR DESCRIPTION
**Pull request by ShourieG**
## Type of change
- Enhancement

## What does this PR do?
Adds support for GCS input to the Cloudflare Logpush package.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues
- Relates #4291

## Screenshots
Screenshot 2022-11-28 at 1 09 39 PM

### CODE REVIEW

File: packages/cloudflare_logpush/data_stream/audit/manifest.yml [hunk 1]
```diff
@@ -149,3 +149,81 @@ streams:
...
         description: >-
           Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
+  - input: gcs
+    title: Audit logs via GCS
+    description: Collect Audit logs from Cloudflare.
+    template_path: gcs.yml.hbs
+    vars:
+      - name: max_workers
+        type: integer
+        title: '[GCS] Maximum number of workers'
```
Comment (efd6) [inline packages/cloudflare_logpush/data_stream/audit/manifest.yml:159]: Do we need the `[GCS]` prefix for these? Perhaps have this information (the association of the config field with the GCS input) in the description.

Comment (ShourieG) [reply packages/cloudflare_logpush/data_stream/audit/manifest.yml:159]: @efd6 actually I did to keep it consistent with the S3 implementation. It uses the [AWS] prefix for all the fields. Other than that there is not reason for this.

**Comment by ShourieG** [comment]
@efd6 have updated the PR

File: packages/cloudflare_logpush/manifest.yml [hunk 2]
```diff
@@ -176,5 +176,39 @@ policy_templates:
...
+            type: text
+            title: "Project Id"
+            description: It is a required parameter to collect logs via GCS.
+            multi: false
+            required: true
+            show_user: true
+            default: my-project-id
+          - name: service_account_key
+            type: password
+            title: "Credentials json key"
```
Comment (efd6) [inline packages/cloudflare_logpush/manifest.yml:201]:
```suggestion
            title: "JSON Credentials key"
```

File: packages/cloudflare_logpush/manifest.yml [hunk 3]
```diff
@@ -176,5 +176,39 @@ policy_templates:
...
+            description: It is an optional parameter for authentication.
+            multi: false
+            required: false
+            show_user: true
+          - name: service_account_file
+            type: text
+            title: "Credentials file path"
```
Comment (efd6) [inline packages/cloudflare_logpush/manifest.yml:208]:
```suggestion
            title: "JSON Credentials file path"
```

File: packages/cloudflare_logpush/manifest.yml [hunk 4]
```diff
@@ -176,5 +176,39 @@ policy_templates:
...
+            title: "Project Id"
+            description: It is a required parameter to collect logs via GCS.
+            multi: false
+            required: true
+            show_user: true
+            default: my-project-id
+          - name: service_account_key
+            type: password
+            title: "Credentials json key"
+            description: It is an optional parameter for authentication.
```
Comment (efd6) [inline packages/cloudflare_logpush/manifest.yml:202]:
```suggestion
            description: The service account key text to authenticate to the store.
```

File: packages/cloudflare_logpush/manifest.yml [hunk 5]
```diff
@@ -176,5 +176,39 @@ policy_templates:
...
+            multi: false
+            required: false
+            show_user: true
+          - name: service_account_file
+            type: text
+            title: "Credentials file path"
+            description: It is an optional parameter for authentication.
```
Comment (efd6) [inline packages/cloudflare_logpush/manifest.yml:209]:
```suggestion
            description: Path to a file containing the service account key text to authenticate to the store.
```

**Review by efd6 — APPROVED** [review]
LGTM after minor change.

**Pull request merged**